src/Tribe/Date_Utils.php

```diff
@@ -1386,7 +1389,7 @@ static function immutable( $datetime = 'now', $timezone = null, $with_fallback =
 			}
 
 			if ( $datetime instanceof DateTime ) {
-				return DateTimeImmutable::createFromMutable( $datetime );
+				return ( new Date_I18n_Immutable() )->setTimestamp( DateTimeImmutable::createFromMutable( $datetime )->getTimestamp() );
```

Comment:
The conversion to timestamp is one that will drop the timezone information defaulting on the use of the system one.
I would rather implement a `Date_I18n_Immutable::createFromMutable` method (in `camelCase` to override the default PHP method and keep the API consistent) where we're never conflating the time information and the timezone information into a timestamp.

Reply:
Nice I like that.